src/webextension/background/authorization.js

```diff
@@ -0,0 +1,214 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import jose from "node-jose";
+
+const OAUTH_CLIENT_ID = "6b7245cb84132a90";
+const OAUTH_CLIENT_SECRET = "bc56d58f7996e5f40db8e1c05f61e745cd4e0664cb4d1f3f787667bcca3e4e79";
+const OAUTH_REDIRECT_URI = "https://2aa95473a5115d5f3deb36bb6875cf76f05e4c4d.extensions.allizom.org/";
+const OAUTH_URL_PREFIX = "https://oauth-stable.dev.lcip.org/v1";
+const OAUTH_PROFILE_URI = "https://stable.dev.lcip.org/profile/v1";
+const OAUTH_SCOPES = ["profile"];
+const FXA_URI_PREFIX = "https://stable.dev.lcip.org/auth/v1";
+
+function toQuery(params) {
+  let query = Object.keys(params).map((k) => {
+    let v = params[k];
+    return `${encodeURIComponent(k)}=${encodeURIComponent(v)}`;
+  });
+  return query.join("&");
+}
+function fromQuery(query) {
+  let params = {};
+  query.substring(1).split("&").forEach((p) => {
+    let [k, v] = p.split("=");
+    params[decodeURIComponent(k)] = decodeURIComponent(v || "");
+  });
+  return params;
+}
+
+async function makeAuthzRequest(state, challenge) {
```

Comment:
Should the Z in `Authz` be capitalized? I see people use it both ways (I don't care much myself).

Reply:
(**bikeshed alert**)

Since `authz` is shorthand for "authorization", `Authz` seems more natural that `AuthZ`.

This is not a hill to die on though...

Reply:
Sounds good to me; I just wanted to be sure we were using a way that people who know what authz is would recognize.